### ISSUE TITLE
[shelly] Device discovery fails, due to NPE

### ISSUE THREAD

**Issue by falkena** [openhab/openhab-addons#7069]
Shelly throws NPE on discovery and floods log file:
```
2020-02-24 22:20:59.277 [WARN ] [discovery.ShellyDiscoveryParticipant] - Device discovery failed for device shelly1pm-XXXXXX, IP <IP>, service=shelly1pm-XXXXXX: null
java.lang.NullPointerException: null
        at org.openhab.binding.shelly.internal.api.ShellyHttpApi.request(ShellyHttpApi.java:474) ~[bundleFile:?]
        at org.openhab.binding.shelly.internal.api.ShellyHttpApi.getDeviceProfile(ShellyHttpApi.java:87) ~[bundleFile:?]
        at org.openhab.binding.shelly.internal.discovery.ShellyDiscoveryParticipant.createResult(ShellyDiscoveryParticipant.java:133) [bundleFile:?]
        at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryService.considerService(MDNSDiscoveryService.java:274) [bundleFile:?]
        at org.eclipse.smarthome.config.discovery.mdns.internal.MDNSDiscoveryService.serviceResolved(MDNSDiscoveryService.java:266) [bundleFile:?]
        at javax.jmdns.impl.ListenerStatus$ServiceListenerStatus.serviceResolved(ListenerStatus.java:106) [bundleFile:3.5.5]
        at javax.jmdns.impl.JmDNSImpl$1.run(JmDNSImpl.java:894) [bundleFile:3.5.5]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_201]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_201]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_201]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_201]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_201]
```

**Comment by markus7017** [comment]
Discovery was re-factored. It seems that there is a mDNS bug, which requires to run the discovery multiple times.

Fix is included in PR #6985

**Comment by markus7017** [comment]
included in PR #6985

**Comment by markus7017** [comment]
included in version 2.5.4
